cmd/odfcli/set/recovery-profile.go

```diff
@@ -0,0 +1,47 @@
+package set
+
+import (
+	"fmt"
+
+	"github.com/red-hat-storage/odf-cli/cmd/odfcli"
+	"github.com/red-hat-storage/odf-cli/pkg/set"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	"github.com/spf13/cobra"
+)
+
+var validRecoveryOptions = []string{"high_client_ops", "high_recovery_ops", "balanced"}
+
+var recoveryProfile = &cobra.Command{
+	Use:                "recovery-profile",
+	Short:              "Set the recovery profile to favor new IO, recovery, or balanced mode with options high_client_ops, high_recovery_ops, or balanced",
+	DisableFlagParsing: true,
+	Args:               cobra.MaximumNArgs(1),
```

Comment:
I have 2 thoughts here:
1. My opinion is that this should be cobra.ExactArgs(1). I don't think the behavior of setting the default when the user gives no setting value matches with how most CLIs work. Users will generally assume that `cli set key` with no value is a typo.
2. Cobra offers more features for setting string arg values that would cut down on the boilerplate and offer other benefits:

Cobra has a `ValidArgs` property that we should use here. This allows us to totally skip the `isValidArg()` boilerplate code, corresponding unit tests, as well as usage of the custom function in the `Command.Run()` method. Pair it with the `MatchAll(ExactArgs(n), OnlyValidArgs)` for validation.

e.g.
```
myCmd := &cobra.Command{
    # ...
    ValidArgs: []string{"balanced", "etc"},
    Args:      cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs)
}
```

FWIW, this would also allow cobra to create auto-completion for the command more easily if we decide to enable it in the future.

Reply:
I agree and I explored the `ValidArgs` but I was not aware of `MatchAll`.